scapy/fields.py

```diff
@@ -1414,7 +1414,11 @@ def i2m(self,
 
     def m2i(self, pkt, m):
         # type: (Optional[Packet], bytes) -> Packet
-        return self.cls(m)
+        if inspect.isclass(self.cls) and \
+           issubclass(self.cls, BasePacket):
+            return self.cls(m, _parent=pkt)
+        else:
+            return self.cls(m)
```

Comment:
Why did you choose a different approach here and in your next code change, to achieve, if I understand well, the same thing?

Reply:
I was fighting type checker, and it was the only way it passed the test.

Reply:
I prefer it indeed, and I think you should use it here too.

Reply:
Changed it